rust/feature-flags/src/flag_matching.rs

```diff
@@ -132,90 +288,144 @@ impl FeatureFlagMatcher {
+            ) {
+                return Ok(local_overrides);
+            }
+        }
+
+        self.get_group_properties_from_in_memory_cache_or_db(feature_flag.team_id, group_type_index)
             .await
-        // TODO handle group properties, will go something like this
-        // if let Some(group_index) = feature_flag.get_group_type_index() {
-        //     self.get_group_properties(feature_flag.team_id, group_index, properties)
-        // } else {
-        //     self.get_person_properties(feature_flag.team_id, properties)
-        //         .await
-        // }
     }
 
-    async fn get_person_properties(
+    async fn get_or_fetch_person_properties(
         &mut self,
-        team_id: i32,
-        properties: &[PropertyFilter],
+        team_id: TeamId,
+        flag_property_filters: &[PropertyFilter],
     ) -> Result<HashMap<String, Value>, FlagError> {
-        if let Some(person_overrides) = &self.person_property_overrides {
-            // Check if all required properties are present in the overrides
-            // and none of them are of type "cohort"
-            let should_prefer_overrides = properties
-                .iter()
-                .all(|prop| person_overrides.contains_key(&prop.key) && prop.prop_type != "cohort");
-
-            if should_prefer_overrides {
-                // TODO let's count how often this happens
-                return Ok(person_overrides.clone());
-            }
+        if let Some(overrides) = self.can_compute_property_overrides_locally(
+            &self.person_property_overrides,
+            flag_property_filters,
+        ) {
+            return Ok(overrides);
         }
 
         // If we don't prefer the overrides (they're either not present, don't contain enough properties to evaluate the condition,
         // or contain a cohort property), fall back to getting properties from cache or DB
-        self.get_person_properties_from_cache_or_db(team_id, self.distinct_id.clone())
+        self.get_person_properties_from_in_memory_cache_or_db(team_id, self.distinct_id.clone())
             .await
     }
 
+    /// Check if all required properties are present in the overrides
+    /// and none of them are of type "cohort" – if so, return the overrides,
+    /// otherwise return None, because we can't locally compute cohort properties
+    fn can_compute_property_overrides_locally(
+        &self,
+        property_overrides: &Option<HashMap<String, Value>>,
+        property_filters: &[PropertyFilter],
+    ) -> Option<HashMap<String, Value>> {
+        property_overrides.as_ref().and_then(|person_overrides| {
+            let should_prefer_overrides = property_filters
+                .iter()
+                .all(|prop| person_overrides.contains_key(&prop.key) && prop.prop_type != "cohort");
+
+            if should_prefer_overrides {
+                Some(person_overrides.clone())
+            } else {
+                None
+            }
+        })
+    }
```

Comment:
just reimplemented the similarly-named method from Django.  It was hard to tell, but I think our code said that both person and group properties needed to obey these constraints, so this method is used in both property fetchers before deciding to use overrides vs fetching DB props.

cc @neilkakkar

Reply:
that's right, same logic applies to both.

One tricky implementation detail here is for matching static cohorts, you'd still need to do this at the db level where you fetch person properties (since we can't do this in python) - so the query there would need to be adjusted to also tell if given person is or isn't in static cohort X, Y, Z.